dagshub/data_engine/model/datasource.py

```diff
@@ -798,16 +926,19 @@ def __getitem__(self, other: Union[slice, str, "Datasource", "Field"]):
         if type(other) is str:
             if not self.has_field(other):
                 raise FieldNotFoundError(other)
-            other_query = DatasourceQuery(other)
-            if self._query.is_empty:
-                new_ds._query = other_query
+            other_query = QueryFilterTree(other)
+            if self._query.filter.is_empty:
+                new_ds._query.filter = other_query
             else:
                 new_ds._query.compose("and", other_query)
             return new_ds
         elif type(other) is Field:
             if not self.has_field(other.field_name):
                 raise FieldNotFoundError(other.field_name)
-            new_ds._query = DatasourceQuery(other.field_name, other.as_of_timestamp)
+            new_ds._query.filter = QueryFilterTree(
+                other.field_name,
+                field_as_of=other.as_of_timestamp,
+            )
```

Comment:
Aren't you missing a `if self._query.filter.is_empty` check here?
If so, and it passed the tests, that means we need more tests for Field scenarios and composition!

Reply:
Amazing catch, thank you.
Added a test that composes string filter -> field filter and vice verse